src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/PageActionInvoker.cs

```diff
@@ -341,6 +341,13 @@ private Task Next(ref State next, ref Scope scope, ref object state, ref bool is
                 _pageContext.ViewData.Model = _model;
             }
 
+            if (CacheEntry.PropertyBinder != null &&
+                !string.Equals(_pageContext.HttpContext.Request.Method, "GET", StringComparison.OrdinalIgnoreCase))
+            {
+                // Don't bind properties on GET requests
```

Comment:
IMO comments on the outskirts of the ifstatement have more value describing the purpose of a piece of code than inside them. Just a personal preference.